src/main/kotlin/org/wycliffeassociates/otter/jvm/persistence/repositories/CollectionRepository.kt

```diff
@@ -238,4 +243,51 @@ class CollectionRepository(
 
         return collectionMapper.mapFromEntity(entity, metadata)
     }
+
+    override fun importResourceContainer(rc: ResourceContainer, rcTree: Tree, languageSlug: String): Completable {
+        return Completable.fromAction {
+            database.transaction { dsl ->
+                val language = languageMapper.mapFromEntity(languageDao.fetchBySlug(languageSlug, dsl))
+                val metadata = rc.manifest.dublinCore.mapToMetadata(rc.dir, language)
+                val metadataId = metadataDao.insert(metadataMapper.mapToEntity(metadata), dsl)
+
+                val root = rcTree.value as Collection
+                val rootEntity = collectionMapper.mapToEntity(root)
+                rootEntity.metadataFk = metadataId
+                val rootId = collectionDao.insert(rootEntity, dsl)
+                for (node in rcTree.children) {
+                    importNode(rootId, metadataId, node, dsl)
+                }
+            }
+        }.subscribeOn(Schedulers.io())
+    }
+
+    private fun importNode(parentId: Int, metadataId: Int, node: TreeNode, dsl: DSLContext) {
+        when(node) {
+            is Tree -> {
+                importCollection(parentId, metadataId, node, dsl)
+            }
+            is TreeNode -> {
+                importChunk(parentId, node, dsl)
+            }
+        }
+    }
+
+    private fun importCollection(parentId: Int, metadataId: Int, node: Tree, dsl: DSLContext){
```

Comment:
If you change parentId to nullable (which is fine since parentFk can be null), you can just call this function in `importResourceContainer` to import the root collection, passing in a null parentId